vue-i18n-bridge: vue-tsc throws TS errors

### Reporting a bug?

```node
node_modules/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts:1:15 - error TS2614: Module '"vue"' has no exported member 'App'. Did you mean to use 'import App from "vue"' instead?

1 import type { App } from 'vue';
                ~~~

node_modules/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts:2:15 - error TS2614: Module '"vue"' has no exported member 'ComponentInternalInstance'. Did you mean to use 'import ComponentInternalInstance from "vue"' instead?

2 import type { ComponentInternalInstance } from 'vue';
                ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts:3:15 - error TS2614: Module '"vue"' has no exported member 'ComputedRef'. Did you mean to use 'import ComputedRef from "vue"' instead?

3 import type { ComputedRef } from 'vue';
                ~~~~~~~~~~~

node_modules/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts:7:10 - error TS2614: Module '"vue"' has no exported member 'InjectionKey'. Did you mean to use 'import InjectionKey from "vue"' instead?

7 import { InjectionKey } from 'vue';
           ~~~~~~~~~~~~

node_modules/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts:45:15 - error TS2614: Module '"vue"' has no exported member 'WritableComputedRef'. Did you mean to use 'import WritableComputedRef from "vue"' instead?

45 import type { WritableComputedRef } from 'vue';
                 ~~~~~~~~~~~~~~~~~~~


Found 5 errors.
```

### Expected behavior

No errors should be thrown

### Reproduction

Set up a simple project with vue 2, vue-tsc and i18n libs

### System Info

```shell
System:
    OS: macOS 12.0.1
    CPU: (8) x64 Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz
    Memory: 217.01 MB / 16.00 GB
    Shell: 5.8 - /usr/local/bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 3.1.0 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 95.0.4638.69
    Firefox: 70.0.1
    Safari: 15.1
  npmPackages:
    @vue/cli-plugin-babel: ~5.0.0-rc.0 => 5.0.0-rc.0 
    @vue/cli-plugin-unit-jest: ~5.0.0-rc.0 => 5.0.0-rc.0 
    @vue/cli-service: ~5.0.0-rc.0 => 5.0.0-rc.0 
    @vue/composition-api: ~1.4.0 => 1.4.0 
    @vue/eslint-config-prettier: ~6.0.0 => 6.0.0 
    @vue/eslint-config-typescript: ~9.0.1 => 9.0.1 
    @vue/runtime-core: ~3.2.22 => 3.2.22 
    @vue/runtime-dom: ~3.2.22 => 3.2.22 
    @vue/test-utils: ~1.2.2 => 1.2.2 
    @vue/vue2-jest: ~27.0.0-alpha.3 => 27.0.0-alpha.3 
    vite: ~2.6.14 => 2.6.14 
    vite-plugin-time-reporter: ~1.0.0 => 1.0.0 
    vite-plugin-ts-nameof: ~1.1.0 => 1.1.0 
    vite-plugin-vue2: ~1.9.0 => 1.9.0 
    vue: ~2.6.14 => 2.6.14 
    vue-chartjs: ~3.5.1 => 3.5.1 
    vue-frag: ~1.3.1 => 1.3.1 
    vue-i18n: ~8.26.7 => 8.26.7 
    vue-i18n-bridge: ~9.2.0-beta.18 => 9.2.0-beta.18 
    vue-moment: ~4.1.0 => 4.1.0 
    vue-resource: ~1.5.3 => 1.5.3 
    vue-router: ~3.5.3 => 3.5.3 
    vue-shortkey: ~3.1.7 => 3.1.7 
    vue-template-compiler: ~2.6.14 => 2.6.14 
    vue-tsc: ~0.29.5 => 0.29.5 
    vuedraggable: ~2.24.3 => 2.24.3 
    vuetify: ~2.5.14 => 2.5.14
```


### Screenshot

![image](https://user-images.githubusercontent.com/7195563/141776728-2a9f7ab1-9909-4db6-9d0e-3de9f769851e.png)


### Additional context

The bridge should maybe use `vue-demi` or just `@vue/composition-api` instead of `vue`

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/intlify/vue-i18n-next/discussions)

--- a/docs/guide/migration/ways.md
+++ b/docs/guide/migration/ways.md
@@ -128,7 +128,8 @@ import { createI18n, useI18n, castToVueI18n } from 'vue-i18n-bridge'
 Vue.use(VueCompositionAPI)
 Vue.use(VueI18n, { bridge: true })
 
-const i18n = createI18n({
+// you need to cast `i18n` instance
+const i18n = castToVueI18n(createI18n({
   locale: 'ja',
   messages: {
     en: {
@@ -142,11 +143,11 @@ const i18n = createI18n({
       }
     }
   }
-}, VueI18n)
+}, VueI18n))
 
 Vue.use(i18n)
 
-const app = new Vue({ i18n: castToVueI18n(i18n) }) // you need to cast `i18n` instance
+const app = new Vue({ i18n })
 app.$mount('#app')
 ```
 

--- a/package.json
+++ b/package.json
@@ -22,7 +22,7 @@
     "build:size-petite-vue-i18n": "cd packages/size-check-petite-vue-i18n && npm run build",
     "build:size-vue-i18n": "cd packages/size-check-vue-i18n && npm run build",
     "build:sourcemap": "npm run build -- --sourcemap",
-    "build:type": "npm run build -- --types && tail -n +32 ./packages/vue-i18n/src/vue.d.ts >> ./packages/vue-i18n/dist/vue-i18n.d.ts && tail -n +32 ./packages/petite-vue-i18n/src/vue.d.ts >> ./packages/petite-vue-i18n/dist/petite-vue-i18n.d.ts && tail -n +32 ./packages/vue-i18n-bridge/src/vue.d.ts >> ./packages/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts",
+    "build:type": "./scripts/build.sh",
     "build:typed": "npm run build -- core-base vue-i18n-core --types",
     "clean": "run-p clean:*",
     "clean:cache": "npm run clean:cache:jest",

--- a/packages/vue-i18n-core/src/i18n.ts
+++ b/packages/vue-i18n-core/src/i18n.ts
@@ -784,11 +784,15 @@ export function useI18n<
  *
  * @VueI18nGeneral
  */
-export const castToVueI18n = /* #__PURE__*/ (i18n: I18n): VueI18n => {
+export const castToVueI18n = /* #__PURE__*/ (
+  i18n: I18n
+): VueI18n & { install: (Vue: any, options?: any) => void } => {
   if (!(__VUE_I18N_BRIDGE__ in i18n)) {
     throw createI18nError(I18nErrorCodes.NOT_COMPATIBLE_LEGACY_VUE_I18N)
   }
-  return i18n as unknown as VueI18n
+  return i18n as unknown as VueI18n & {
+    install: (Vue: any, options?: any) => void
+  }
 }
 
 function createGlobal(

--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -1,0 +1,12 @@
+#!/bin/bash
+
+set -xe
+
+
+npm run build -- --types
+
+tail -n +32 ./packages/vue-i18n/src/vue.d.ts >> ./packages/vue-i18n/dist/vue-i18n.d.ts
+tail -n +32 ./packages/petite-vue-i18n/src/vue.d.ts >> ./packages/petite-vue-i18n/dist/petite-vue-i18n.d.ts
+tail -n +32 ./packages/vue-i18n-bridge/src/vue.d.ts >> ./packages/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts
+
+node -r esbuild-register scripts/postprocess.ts

--- a/scripts/postprocess.ts
+++ b/scripts/postprocess.ts
@@ -1,0 +1,43 @@
+import path from 'pathe'
+import { promises as fs } from 'fs'
+
+function replaceWithCompositionApi(source: string, target: string) {
+  return source.replace(
+    `{ ${target} } from 'vue';`,
+    `{ ${target} } from '@vue/composition-api';`
+  )
+}
+
+async function main() {
+  let source = await fs.readFile(
+    path.resolve(
+      __dirname,
+      '../packages/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts'
+    ),
+    'utf8'
+  )
+
+  source = [
+    'App',
+    'ComponentInternalInstance',
+    'ComputedRef',
+    'InjectionKey',
+    'WritableComputedRef'
+  ].reduce(
+    (source, target) => replaceWithCompositionApi(source, target),
+    source
+  )
+
+  await fs.writeFile(
+    path.resolve(
+      __dirname,
+      '../packages/vue-i18n-bridge/dist/vue-i18n-bridge.d.ts'
+    ),
+    source
+  )
+}
+
+main().catch(err => {
+  console.error(err)
+  process.exit(1)
+})

